### PR TITLE
Make ServiceAuthenticationManager async

### PR DESCRIPTION
**Pull request by g7ed6e**

### CODE REVIEW

**Comment by mconnew** [comment]
Same issue as the other PR for ServiceAuthorizationManager, the default implementation needs to stay in the sync methods and have the async methods call them.

**Comment by g7ed6e** [comment]
> Same issue as the other PR for ServiceAuthorizationManager, the default implementation needs to stay in the sync methods and have the async methods call them.

I inverted the delegation to ensure compatibility with existing implementations.

File: src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs [hunk 1]
```diff
@@ -45,7 +53,7 @@ public override ReadOnlyCollection<IAuthorizationPolicy> Authenticate(ReadOnlyCo
                 authPolicy = authPolicies.AsReadOnly();
             }
 
-            return _wrappedAuthenticationManager.Authenticate(authPolicy, listenUri, ref message);
+            return await _wrappedAuthenticationManager.AuthenticateAsync(authPolicy, listenUri, message);
         }
```
Comment (mconnew) [inline src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs:57]: When the only await is the last line of the method, you don't need to make the method async. You can directly return the ValueTask from _wrappedAuthenticatonManager

File: src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs [hunk 2]
```diff
@@ -84,18 +92,18 @@ public override ReadOnlyCollection<IAuthorizationPolicy> Authenticate(ReadOnlyCo
...
                     {
-                        return authPolicy;
+                        return (authPolicy, message);
                     }
                 }
             }
 
-            return _wrappedAuthenticationManager.Authenticate(authPolicy, listenUri, ref message);
+            return await _wrappedAuthenticationManager.AuthenticateAsync(authPolicy, listenUri, message);
         }
```
Comment (mconnew) [inline src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs:101]: Same as above with not needing to use await

File: src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs (hunk 1 above)
Comment (g7ed6e) [reply src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthenticationManager.cs:57]: i removed the async /await

**Review by mconnew — APPROVED** [review]

**Pull request merged**